cms/djangoapps/contentstore/rest_api/v1/views/tests/test_xblock.py

```diff
@@ -0,0 +1,291 @@
+from unittest.mock import patch
+from django.http import JsonResponse
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+from cms.djangoapps.contentstore.tests.test_utils import AuthorizeStaffTestCase
+
+
+TEST_LOCATOR = "block-v1:dede+aba+weagi+type@problem+block@ba6327f840da49289fb27a9243913478"
+
+
+class XblockViewTestCase(AuthorizeStaffTestCase):
+    # assumes that you want to pass a block id to the url
+
+    def get_test_data(self):
+        raise NotImplementedError("get_test_data must be implemented by subclasses")
+
+    def get_url_params(self):
+        return {"course_id": self.get_course_key_string(), "usage_key_string": TEST_LOCATOR}
+
+    def get_url(self):
+        return reverse(
+            "cms.djangoapps.contentstore:v1:studio_content",
+            kwargs=self.get_url_params(),
+        )
+
+    def send_request(self):
+        raise NotImplementedError("send_request must be implemented by subclasses")
+
+    @patch(
+        "cms.djangoapps.contentstore.rest_api.v1.views.xblock.handle_xblock",
+        return_value=JsonResponse(
+            {
+                "locator": TEST_LOCATOR,
+                "courseKey": AuthorizeStaffTestCase.get_course_key_string(),
+            }
+        ),
+    )
+    @patch(
+        "cms.djangoapps.contentstore.rest_api.v1.views.xblock.toggles.use_studio_content_api",
+        return_value=True,
+    )
+    def make_request(
+        self,
+        mock_use_studio_content_api,
+        mock_handle_xblock,
+        run_assertions=None,
+        course_id=None,
+        data=None,
+    ):
+        url = self.get_url()
```

Comment:
I recommend adding the following comment here, or something like it:

        """
        Note that the actual xblock handler is mocked out and not used here. Patches used with this method serve to
        test that routing of HTTP requests to the xblock handler is correct, that the intended HTTP method has been
        used, that data fed into the handler is as expected, and that data returned by the handler is as expected.

        Inputs and outputs are handled through send_request() polymorphism, to cover all the HTTP methods in a
        common fashion here.

        Validations are through injection of run_assersions().

        """